src/rust/ide/lib/view/src/graph_editor/component/node/port.rs

```diff
@@ -285,9 +314,21 @@ impl Manager {
         }).ok()
     }
 
+    pub fn get_port_color(&self, crumbs:&[span_tree::Crumb]) -> Option<color::Lcha> {
+        let ast_id = self.expression.borrow().get_id_for_crumbs(crumbs)?;
+        self.type_color_map.type_color(ast_id)
+    }
+
     pub fn width(&self) -> f32 {
         self.width.get()
     }
+
+    pub fn set_expression_type(&self, id:ast::Id, maybe_type:Option<Type>) {
+        match maybe_type {
+            Some(r#type) => self.type_color_map.insert(id,r#type),
```

Comment:
Please fix all such naming in the PR . `r#type` is really not the best name for the var. In case you don't agree,l sets chat about it! But I was writing about it previously already :)